contentcuration/contentcuration/viewsets/channel.py

```diff
@@ -639,26 +641,44 @@ def publish_next(self, pk):
                     progress_tracker=progress_tracker,
                     use_staging_tree=True,
                 )
-                Change.create_changes(
-                    [
-                        generate_update_event(
-                            channel.id,
-                            CHANNEL,
-                            {
-                                "primary_token": channel.get_human_token().token,
-                            },
-                            channel_id=channel.id,
-                        ),
-                    ],
+                Change.create_change(
+                    generate_update_event(
+                        channel.id,
+                        CHANNEL,
+                        {
+                            "primary_token": channel.get_human_token().token,
+                            "staging_publishing": False,
+                        },
+                        channel_id=channel.id,
+                    ),
                     applied=True,
+                    created_by_id=self.request.user.id,
                 )
             except ChannelIncompleteError:
-                channel.staging_tree.publishing = False
-                channel.staging_tree.save()
+                Change.create_change(
+                    generate_update_event(
+                        channel.id,
+                        CHANNEL,
+                        {"staging_publishing": False},
+                        channel_id=channel.id,
+                    ),
+                    applied=True,
+                    unpublishable=True,
+                    created_by_id=self.request.user.id,
+                )
                 raise ValidationError("Channel is not ready to be published")
             except Exception:
-                channel.staging_tree.publishing = False
-                channel.staging_tree.save()
+                Change.create_changes(
```

Comment:
(Just nitpick :sweat_smile:) Could we move this duplicated logic for handling publishing failure into another function so we can use it in both exceptions? I see the only difference is that in the `ChannelIncompleteError` we are raising a ValidationError and here not.